docs/docs/adding-search-with-algolia.md

```diff
@@ -2,593 +2,579 @@
 
-![Algolia API key screenshot](./images/algolia-api-keys.png)
+The configuration could have been entered straight into the `gatsby-config.js`, but the configuration above loads it from a new file `src/utils/algolia-queries.js` to avoid clutter. Create this page in your project:
 
-Once you have your App ID, Search-Only API Key, and Admin API Key, place the following code into your `.env` file, replacing the placeholder keys with your keys:
-
-```text:title=.env
-GATSBY_ALGOLIA_APP_ID=KA4OJA9KAS
-GATSBY_ALGOLIA_SEARCH_KEY=lkjas987ef923ohli9asj213k12n59ad
-ALGOLIA_ADMIN_KEY=lksa09sadkj1230asd09dfvj12309ajl
-```
-
-The placeholder keys in the previous code snippet are random character sequences but the ones you copy from your Algolia profile should be the same length. One of the benefits of using this method of querying your API keys is that they all get stored in one file, on the server, and are therefore never exposed to the client-side, which increases security.
-
-Since your .env file contains your real private API keys, it is considered a security risk to commit your actual `.env` file. It's good practice to commit a `.env.example` to git or other version control so that if someone forks your repo, they know which environment variables they need to supply, without committing your private keys.
-
-```text:title=.env.example
-# rename this file to .env and supply the values listed below
-# also make sure they are available to the build tool (e.g. Netlify)
-# warning: variables prefixed with GATSBY_ will be made available to client-side code
-# be careful not to expose sensitive data (in this case your Algolia admin key)
-
-GATSBY_ALGOLIA_APP_ID=insertValue
-GATSBY_ALGOLIA_SEARCH_KEY=insertValue
-ALGOLIA_ADMIN_KEY=insertValue
-```
-
-The `queries` allow you to grab the data you want Algolia to index directly from Gatsby's GraphQL layer by exporting from `src/utils/algolia.js` an array of objects, each containing a required GraphQL query and an optional index name, transformer function and settings object.
-
-> Note: The excerpts below are pruned to 5000 characters. This is because, by default, Algolia has an upper bound of 10KB for an index entry. If you hit this boundary, you'll be presented with an error like this: `AlgoliaSearchError: Record at the position XX objectID=xx-xx-xx-xx-xx is too big size=10356 bytes. Contact us if you need an extended quota`. Make sure that your overall record is within this limit.
-
-```js:title=src/utils/algolia.js
+```js:title=src/utils/algolia-queries.js
 const pageQuery = `{
   pages: allMarkdownRemark(
     filter: {
-      fileAbsolutePath: { regex: "/pages/" },
-      frontmatter: {purpose: {eq: "page"}}
+      fileAbsolutePath: { regex: "/${pagePath.replace("/", "/")}/" },
```

Comment:
This throws an error, since there is no `pagePath` defined.

Reply:
Sorry, I managed to lose the two first lines with constants during the rewrite to work with starter-blog. I added them back now. Anyway, at the beginning of the file it should say `const pagePath = "content"

Reply:
`.replace("/", "/")`

what is the meaning of this? I think it not changes the value?

Did you mean this?

```suggestion
      fileAbsolutePath: { regex: "/${pagePath}/".replace("//", "/") },
```

Reply:
No that's of course nonsense. I also just noticed. I think the backslash got lost when I copied the code back and forth. 

I meant `regex: "/${pagePath.replace("/", "\\/")}/"` The idea is that the regexp should work correctly even if the path contains a forward slash.

Reply:
Is there not a Gatsby until or a npm package for unifying this kind of slashes? 

I expected more a removing for double slashes if pagePath already includes slashes at start or at the end

Reply:
even the it should be?

```javascript
"/${pagePath.replace("\\", "/")}/"
```

Reply:
If you do this to escape the slash then I remember you need only to escape it with `\/` when you assign the regex direct:

```javascript
const regex = /path\/file/ig 
```

But if it is in a string then you don’t need to escape and don’t include the slash at start and end:

```javascript 
const regexstring = "path/file"
const regex = new RegExp(regexstring)
```

i am a little confused. Maybe I am wrong 🤷‍♂️

Reply:
it seems i am wrong: i looked only at the diff and i did not see that this code snippet is inside a graphql query 🤦

Reply:
Yeah, it's a mess with these double and triple quotes. I found the library you mentioned (`escape-string-regexp`) and added it instead. Using it makes more explicit what the code is trying to achieve and in case someone names their blog directory `[a-z]` it will still work...

However, I also noticed after a bit of head scratching that `packages/gatsby/src/utils/prepare-regex.ts` works both with double quoting and without (though searching for double backslashes is probably broken :D). So just dropping in the raw `pagePath` works, though it looks broken.